feat(tableHelper): add getEntryKeyByValue

HotaruBlaze opened the pull request:
This is a kinda lazy way but only working way I found of handling #42, I initially tried to make it return the table to be modified, but it does not seem to work. So the closest i got was returning a matching key on match, and using that for the table.

Example of how i'm using it below.
```lua

local playerVoidStorageName = tableHelper.getEntryKeyByValue(WorldInstance.data.voidStorage, playerName)
if not playerVoidStorageName then
	WorldInstance.data.voidStorage[playerName] = { storageId = "", hasAccess = {} }
	playerVoidStorageName = playerName
else

local storageId = WorldInstance.data.voidStorage[playerVoidStorageName].storageId
```
---
Issue #46: Accidently pushed the non-working broken function attempt of returning a table instead of the key, aswell as named the branch the bad function pushed.

YvanCywan commented on lib/lua/tableHelper.lua:
```diff
@@ -400,6 +400,19 @@ function tableHelper.isEmpty(inputTable)
+-- Note: This function is intended to handle situations where the loaded value may not align with the expected format due to the behavior of "cjson."
+function tableHelper.getEntryKeyByValue(inputTable, searchValue)
+    for key, value in pairs(inputTable) do
+        if key == searchValue then
+            return key
+        elseif tostring(key) == tostring(searchValue) then
+            return key
+        end
```
```suggestion
        if key == searchValue or tostring(key) == tostring(searchValue) then
            return key
        end
```

YvanCywan commented on lib/lua/tableHelper.lua:
```diff
@@ -400,6 +400,19 @@ function tableHelper.isEmpty(inputTable)
+            return key
+        end
+    end
+    return nil
```
Considering that this is new code, is it really the best practice to return explicit nils?
Surely we could return false here and it'd have a more descriptive effect?

YvanCywan commented on lib/lua/tableHelper.lua:
```diff
@@ -400,6 +400,19 @@ function tableHelper.isEmpty(inputTable)
 
+-- Retrieve a key associated with a specific value within a table.
+-- Note: This function is intended to handle situations where the loaded value may not align with the expected format due to the behavior of "cjson."
+function tableHelper.getEntryKeyByValue(inputTable, searchValue)
```
It'd be best practice to include annotations on new functions you create.

HotaruBlaze committed:
feat(tableHelper): add getEntryKeyByValue

HotaruBlaze commented:
Addressed the comments, just did a force push to reuse the commit message cuz I'm still waking up. should all be good now, tested it to make sure it still works fine.

YvanCywan commented on lib/lua/tableHelper.lua:
```diff
@@ -400,6 +400,21 @@ function tableHelper.isEmpty(inputTable)
+--- Retrieve a key associated with a specific value within a table.
+---
+--- @param inputTable table  The table to search within.
+--- @param searchValue any  The value to search for within the table.
+--- @return any|false  The key associated with the searchValue if found, or false if not found.
```
The key cannot be any data type
```suggestion
--- @param searchValue string|number  The value to search for within the table.
--- @return string|number|false  The key associated with the searchValue if found, or false if not found.
```

HotaruBlaze committed:
Update lib/lua/tableHelper.lua

Co-authored-by: Yvan Cywan <c1yvan@outlook.com>
